c10/core/TensorImpl.cpp

```diff
@@ -85,7 +86,7 @@ TensorImpl::TensorImpl(Storage&& storage, DispatchKeySet key_set, const caffe2::
   // a backend DispatchKey and an AutogradBackend key.
   // We automatically add the corresponding autograd key to key_set_ so that backends can stay
   // in the old way of only registering with backend key like DispatchKey::CPU.
-  if (c10::InferenceMode::is_enabled()) {
+  if (is_view || c10::InferenceMode::is_enabled()) {
```

Comment:
This logic can be moved to as_view, will come back to this later.

Reply:
After thinking for a while I decided to keep this logic in constructor for now. One main reason is to keep dispatch_key manipulation of tensor all in one place and local to tensor constructor instead of all over the codebase. But I'm definitely open to suggestions about it.